src/SqlSafetyClassifier.php

```
@@ -0,0 +1,100 @@
+        $withoutHashComments = preg_replace('/#.*$/m', ' ', $withoutLineComments) ?? $withoutLineComments;
+        $collapsed = preg_replace('/\s+/', ' ', $withoutHashComments) ?? $withoutHashComments;
+
+        return trim($collapsed);
+    }
+
+    /** @psalm-pure */
+    private function containsUnsafeReadSideEffect(string $normalizedSql): bool
+    {
+        $lower = strtolower($normalizedSql);
+
+        return preg_match('/\bfor\s+update\b/i', $lower) === 1
```

PHPStan (2.2.1)

[error] Possibly impure call to function preg_match() in pure method Koriym\SqlQuality\SqlSafetyClassifier::containsUnsafeReadSideEffect().


(possiblyImpure.functionCall)